ivy/functional/backends/paddle/statistical.py

```diff
@@ -77,9 +77,15 @@ def max(
         paddle.bool,
     ]:
         if paddle.is_complex(x):
-            real_part = paddle.amax(x.real(), axis=axis, keepdim=keepdims)
-            imag_part = paddle.amax(x.imag(), axis=axis, keepdim=keepdims)
-            ret = paddle.complex(real_part, imag_part)
+            real = x.real()
+            imag = x.imag()
+            const = paddle.to_tensor(1j, dtype=x.dtype)
+            real_max = paddle.max(real, axis=axis, keepdim=keepdims)
+            imag = paddle.where(real == real_max, imag, paddle.full_like(imag, -1e5))
```

Comment:
I'm not sure I like this method since the maximum imaginary part could still be less than `-1e5` (in `complex64` it does down to about `-1e38`). I also don't really like the idea of defining complex number behaviour for the `max` function since it's technically undefined mathematically.

It seems like the main reason to implement this here is to use it in softmax for better stability, but in that case the imaginary part doesn't really matter. Might be better to revert the change to this file and just put a simple `paddle.max(x.real(), ...)` in the `softmax` function

Reply:
the imaginary part is important when we the real parts are equal suppose we have 
```python
# in this example the real parts are equal and we define the bigger between these two according to imaginary parts
x = np.array([1-1j, 1+1j])
```

Reply:
After some reverse engineering I discovered how does **numpy** and **jax** support complex **dtype** as you can see first we take the maximum according to the real part  but as I said before we could encounter something like mentioned above where the real parts are equal so the suggested solution works fine like **numpy** and **jax** and the -1e6 problem I couldn't find a way in paddle to get the minimum number of the **dtype** automatically so I put this number until I find a way to determine the minimum  number in a certain dtype.